tests/conftest.py

```
@@ -0,0 +1,379 @@
+    Example:
+        @pytest.mark.parametrize("device_name", ALL_MOBILE_DEVICES)
+        def test_mobile(mobile_page, device_name):
+            mobile_page.goto(BASE_URL)
+            # ... test mobile-specific behavior
+    """
+    # Get device_name from parametrize (safely handle missing callspec)
+    callspec = getattr(request.node, "callspec", None)
+    if callspec is None:
+        raise ValueError(
+            "mobile_page fixture requires @pytest.mark.parametrize with 'device_name' parameter"
+        )
```

Ruff (0.14.11)

Avoid specifying long messages outside the exception class

(TRY003)